core/station/impl/src/services/system.rs

```diff
@@ -548,28 +582,41 @@ mod install_canister_handlers {
     }
 
     /// Starts the fund manager service setting it up to monitor the upgrader canister cycles and top it up if needed.
-    pub fn monitor_upgrader_cycles(upgrader_id: Principal) {
+    pub fn monitor_upgrader_cycles(
+        upgrader_id: Principal,
+        cycle_obtain_strategy: Option<CycleObtainStrategy>,
+    ) {
         print(format!(
-            "Starting fund manager to monitor upgrader canister {} cycles",
+            "Starting fund manager to monitor self {} and upgrader canister {} cycles",
+            id(),
             upgrader_id.to_text()
         ));
 
         FUND_MANAGER.with(|fund_manager| {
             let mut fund_manager = fund_manager.borrow_mut();
 
-            fund_manager.with_options(
-                FundManagerOptions::new()
-                    .with_interval_secs(24 * 60 * 60) // daily
-                    .with_strategy(FundStrategy::BelowEstimatedRuntime(
-                        EstimatedRuntime::new()
-                            .with_min_runtime_secs(14 * 24 * 60 * 60) // 14 days
-                            .with_fund_runtime_secs(30 * 24 * 60 * 60) // 30 days
-                            .with_max_runtime_cycles_fund(1_000_000_000_000)
-                            .with_fallback_min_cycles(125_000_000_000)
-                            .with_fallback_fund_cycles(250_000_000_000),
-                    )),
-            );
-            fund_manager.with_cycles_fetcher(Arc::new(FetchCyclesBalanceFromCanisterStatus));
+            let mut fund_manager_options = FundManagerOptions::new()
+                .with_interval_secs(24 * 60 * 60) // daily
+                .with_strategy(FundStrategy::BelowEstimatedRuntime(
+                    EstimatedRuntime::new()
+                        .with_min_runtime_secs(14 * 24 * 60 * 60) // 14 days
+                        .with_fund_runtime_secs(30 * 24 * 60 * 60) // 30 days
+                        .with_max_runtime_cycles_fund(1_000_000_000_000)
+                        .with_fallback_min_cycles(125_000_000_000)
+                        .with_fallback_fund_cycles(250_000_000_000),
+                ));
+
+            if let Some(strategy) = cycle_obtain_strategy {
+                fund_manager_options = fund_manager_options
+                    .with_obtain_cycles(SYSTEM_SERVICE.get_obtain_cycle_config(&strategy));
+            }
+
+            fund_manager.with_options(fund_manager_options);
+            fund_manager.with_cycles_fetcher(Arc::new(FetchCyclesBalanceFromCanisterStatus {}));
```

Comment:
this would fail on `mainnet`, because the station is not a controller of itself and due to that it can't fetch it's own `canister_status`, would be great to add an integration test that validates this.

Reply:
See commit #[51cc892](https://github.com/dfinity/orbit/pull/307/commits/51cc8921647f5b465cfe8bf1a8784a45f918c25f)